### PR TITLE
send post req to user-api with superagent

### PR DESCRIPTION
**Pull request by LeonardLT**
Now,this can send a post req to userApi with superagent.
if success,it will alert"success", and log 201 ,message：‘register success.’
if error,it will send 400/409,but can not show "error message"。

### CODE REVIEW

**Comment by xiayu0321** [comment]
:+1:

File: app/api/api.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import express from 'express';
+import userApi from './userApi'
```
Comment (freewind) [inline app/api/api.js:2]: user-api.js

File: app/api/userApi.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import express from 'express';
+import {User} from '../db/schema';
+
+
+const router = express.Router();
+
+function getUserInformation(req) {
+    const data = req.body;
+    return data;
+}
```
Comment (freewind) [inline app/api/userApi.js:10]: 不需要抽成函数

File: app/api/userApi.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+function isUserInformationLegal(data) {
+    const regEmail = /^[a-z0-9]+([._\\-]*[a-z0-9])*@([a-z0-9]+[-a-z0-9]*[a-z0-9]+.){1,63}[a-z0-9]+$/;
```
Comment (freewind) [inline app/api/userApi.js:13]: /shared/user-field-validation.js
- validateEmail()
- validatePhone()

File: app/api/userApi.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    const regPhone = /^(\+86)?(1[0-9]{10})$/;
+
+    console.log(regEmail.test(data.email));
+    console.log(regPhone.test(data.phone) + 'Phone');
```
Comment (freewind) [inline app/api/userApi.js:17]: delete

File: app/api/userApi.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        console.log('--data is null--');
+        return {legal: false, message: 'Please finish the form'};
+    }
+    else if (regEmail.test(data.email) == false) {
+        return {legal: false, message: 'The email is error'};
+    }
+    else if (data.phone.length != 11 || regPhone.test(data.phone) == false) {
+        return {legal: false, message: 'The phone number is error'};
+    }
+    else return {legal: true, message: 'legal is true'};
```
Comment (freewind) [inline app/api/userApi.js:30]: 这里应该写成几个独立的方法判断

File: app/api/userApi.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+    if (data.name == '' || data.password == '' || data.email == '' || data.phone == '') {
+        console.log(data);
+        console.log('--data is null--');
+        return {legal: false, message: 'Please finish the form'};
+    }
+    else if (regEmail.test(data.email) == false) {
+        return {legal: false, message: 'The email is error'};
+    }
+    else if (data.phone.length != 11 || regPhone.test(data.phone) == false) {
```
Comment (freewind) [inline app/api/userApi.js:27]: phone的判断有问题

File: app/api/userApi.js (hunk 3 above)
Comment (freewind) [inline app/api/userApi.js:13]: 只返回Boolean

File: app/api/userApi.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        return {legal: false, message: 'The phone number is error'};
+    }
+    else return {legal: true, message: 'legal is true'};
+}
+
+
+router.post('/', function (req, res, next) {
+    debugger;
```
Comment (freewind) [inline app/api/userApi.js:35]: ...

File: app/api/userApi.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    getUserInformation(req);
```
Comment (freewind) [inline app/api/userApi.js:36]: ???

File: app/api/userApi.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    const data = getUserInformation(req);
+    console.log('data:' + data.name);
+    const legal = isUserInformationLegal(data);
+    console.log('legal:' + legal);
+    console.log(legal.message);
+    if (legal.legal == true) {
```
Comment (freewind) [inline app/api/userApi.js:42]: ??? == true

Comment (freewind) [inline app/api/userApi.js:42]: if(!legal.legal) {
   return  res.status(400).send(legal.message);
}

File: app/api/userApi.js [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        return {legal: false, message: 'The phone number is error'};
+    }
+    else return {legal: true, message: 'legal is true'};
+}
+
+
+router.post('/', function (req, res, next) {
+    debugger;
+    getUserInformation(req);
+    const data = getUserInformation(req);
```
Comment (freewind) [inline app/api/userApi.js:37]: userData

File: app/api/userApi.js [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    console.log('data:' + data.name);
+    const legal = isUserInformationLegal(data);
+    console.log('legal:' + legal);
+    console.log(legal.message);
+    if (legal.legal == true) {
+        User.findOne({name: data.name}, function (err, docs) {
```
Comment (freewind) [inline app/api/userApi.js:43]: docs? => user/foundUser

File: app/api/userApi.js [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+                });
+                user.save(function (err) {
+                    console.log('save status:', err ? 'failed' : 'success');
+                    res.status(201).send('register success');
+
+                });
+            }
+            else if (docs != null) {
+
+                res.status(409).send('Same name in db.');
```
Comment (freewind) [inline app/api/userApi.js:61]: 提到前面先判断

File: app/api/userApi.js [hunk 13]
```diff
@@ -0,0 +1,76 @@
+import express from 'express';
+import {User} from '../db/schema';
+
+
+const router = express.Router();
+
+function getUserInformation(req) {
+    const data = req.body;
+    return data;
```
Comment (freewind) [inline app/api/userApi.js:9]: const { name, password, email, phone} = req.body
return { name, password, email, phone}

File: app/api/userApi.js [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+        User.findOne({name: data.name}, function (err, docs) {
+            if (err) return next(err);
+
+            if (docs == null) {
+                var user = new User({
+                    name: data.name,
+                    password: data.password,
+                    email: data.email,
+                    phone: data.phone
+                });
```
Comment (freewind) [inline app/api/userApi.js:52]: const user = new User(userData)

File: app/api/userApi.js [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+    console.log('data:' + data.name);
+    const legal = isUserInformationLegal(data);
+    console.log('legal:' + legal);
+    console.log(legal.message);
+    if (legal.legal == true) {
+        User.findOne({name: data.name}, function (err, docs) {
+            if (err) return next(err);
+
+            if (docs == null) {
+                var user = new User({
```
Comment (freewind) [inline app/api/userApi.js:47]: var?

File: app/api/userApi.js [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+                    name: data.name,
+                    password: data.password,
+                    email: data.email,
+                    phone: data.phone
+                });
+                user.save(function (err) {
+                    console.log('save status:', err ? 'failed' : 'success');
```
Comment (freewind) [inline app/api/userApi.js:54]: err handling ??

File: app/api/userApi.js (hunk 12 above)
Comment (freewind) [inline app/api/userApi.js:61]: "Same name in db." -> username exists

File: app/db/login-and-register.js [hunk 17]
```diff
@@ -36,7 +36,6 @@ exports.login = function (req, res) {
         if (err) res.send(err.message);
 
         if (docs != null) {
-            // res.send('success');
             res.json({name: docs.name, type: 'success'});
         } else {
             res.send('error');
```
Comment (freewind) [inline app/db/login-and-register.js:41]: delete this file?

File: app/server.js [hunk 18]
```diff
@@ -31,6 +32,8 @@ app.use(webpackHotMiddleware(compiler, {
 
 app.use(express.static('./public'));
 
+app.use('/api', apiRouter);
+
 app.get('/hello', function (req, res) {
     res.send('Index, world!');
 });
```
Comment (freewind) [inline app/server.js:39]: can delete /hello

File: public/component/register.jsx [hunk 19]
```diff
@@ -1,71 +1,116 @@
...
-                }
-            },
-            error(result){
-                if (result == "error") {
-                    alert('用户已存在');
-                }
-            }
-        });
+import React, {Component} from "react";
+import request from 'superagent';
```
Comment (freewind) [inline public/component/register.jsx:2]: 加一行空行

File: public/component/register.jsx [hunk 20]
```diff
@@ -1,71 +1,116 @@
...
+                console.log(res.text);
+                // alert("success!");
+
+            });
+        event.preventDefault();
+    }
 }
 
-export default Register;
+
```
Comment (freewind) [inline public/component/register.jsx:116]: 这个文件不是最新的，不看

**Pull request merged**